R/cast.R

```diff
@@ -165,11 +160,28 @@ vec_coercible_cast <- function(x, to, ..., x_arg = "", to_arg = "") {
 #' @inheritParams vec_cast
 #' @export
 vec_default_cast <- function(x, to, x_arg = "", to_arg = "") {
+  if (is_bare_list(to)) {
```

Comment:
Does this have any performance implications?

Reply:
I imagine this will get removed when we restrict `vec_cast` to only common types?

Is that also true for the `vctr_cast()` and `vec_cast_to_list_of()` calls below?

Reply:
Yes this branch and the list-of branch are temporary.

I'd like to remove the `vctrs_vctr` branch but it's a bit more tricky. The class tries to be helpful in more cases in a way that I don't fully understand.